### ISSUE TITLE
Allow adding label to receive addresses

### ISSUE THREAD

**Issue by johnwc** [BlueWallet/BlueWallet#3313]
It would be nice if we could add short label to receive addresses so that when viewing transactions we can see what it was from.

**Comment by ncoelho** [comment]
Go to your transactions details, there's an input for labels. Same on coin control there's also labels there.

**Comment by johnwc** [comment]
I'm not referring to labeling each transaction as it comes in. I am referring to how other wallets allow you to label a receive address, so that when transaction show up that have that label on them automatically.

**Comment by ncoelho** [comment]
Receive with amount > description

See if it works.

**Comment by johnwc** [comment]
This is a watch-only wallet.

**Comment by marcosrdz** [comment]
You can assign a note to transactions via transaction details, or coin control. You can also provide a note and amount in the receive screen, which you can then share.

**Comment by johnwc** [comment]
Again, this is a watch-only account, I have no receive screen. I am not talking about labeling a transaction after it is appeared. I am referring to giving receive addresses labels. So, that when transactions do show up using one that is labeled, they show up in the transaction list labeled already.

**Comment by ncoelho** [comment]
A watch-only wallet only has a receive screen.

**Comment by johnwc** [comment]
These are ongoing transactions to the same receive address.

**Comment by marcosrdz** [comment]
> These are ongoing transactions to the same receive address.

Setting a note for different transactions to the same address would not be possible. This is because the address is being reused, thus making the previous note irrelevant. This is why notes are applied to transactions and not addresses.

**Comment by johnwc** [comment]
I'm not talking about a note... I'm referring to a feature request to have a transaction displayed with a defined label, if the matching receive address in the transaction also has a label applied. Almost all other wallets do this. When creating a new receive address, you can set a label. And from that point forward when a transaction pops up having that address in it's list of addresses, it shows that label.

**Comment by Overtorment** [comment]
Makes sense. Imagine getting and address and putting on some payout page. In 12 months you won’t understand what’s this incoming transaction is. Unless you label the address in advance

**Comment by yashrajd** [comment]
I know you folks already know (as one of the first wallets to support silent payments ❤️) this, but this is precisely the kinda thing [labels](https://github.com/bitcoin/bips/blob/532c4c10f2e04b3dca7d39ce1b1a4bdbf0c88e52/bip-0352.mediawiki?plain=1#L113) in BIP-352 (silent payments) was designed to do.... Implementing silent payments Receive support fully will address this 🫡🫡

The Bitcoin Design Community recently did a [UX exploration & explainer of silent payments](https://bitcoin.design/guide/how-it-works/silent-payments/#tracking-payments) where we covered these use cases. Would love to see address labels 'addressed' :)
